### PR TITLE
Update jsonpatch.md

### PR DESCRIPTION
**Pull request by Rick-Anderson**
REview ulr
https://review.docs.microsoft.com/en-us/aspnet/core/web-api/jsonpatch?view=aspnetcore-3.0&branch=pr-en-us-15404#jsonpatch-addnewtonsoftjson-and-systemtextjson

### CODE REVIEW

File: aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var jsonPatchServices = new ServiceCollection()
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: @Paul-Dempsey  can you help me fix this code. It doesn't compile. I did add  Paul-Dempsey

Comment (Paul-Dempsey) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Never done this before -- working on getting it pulled down to give it whirl...

Comment (serpent5) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Let me know if I can help with this.

Comment (Paul-Dempsey) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Now I'm stuck at:
```
C:\repos\AspNetCore.Docs>git checkout -b Paul-Dempsey-patch-3 54c9efa25ac440f0722a322e030f25e4e4a4bac454c9efa25ac440f0722a322e030f25e4e4a4bac4
fatal: '54c9efa25ac440f0722a322e030f25e4e4a4bac454c9efa25ac440f0722a322e030f25e4e4a4bac4' is not a commit and a branch 'Paul-Dempsey-patch-3' cannot be created from it

C:\repos\AspNetCore.Docs>git checkout -b Paul-Dempsey-patch-3 origin/aspnet/Rick-Anderson-Json-Txt-patch-3
fatal: 'origin/aspnet/Rick-Anderson-Json-Txt-patch-3' is not a commit and a branch 'Paul-Dempsey-patch-3' cannot be created from it
```

Comment (Rick-Anderson) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Just edit in the browser. GitHub will take care of everything for you.

Comment (Paul-Dempsey) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: well if I'm going to fix a compile error, I need to compile it., and I prefer doing that kind of thing in Visual Studio/VSCode

Comment (Rick-Anderson) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Just edit any file in browser and create a PR.. That will make a branch for you. Then check out your branch.
Indirect but it works.

Comment (Rick-Anderson) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Or select my branch from the branch picker
https://github.com/aspnet/AspNetCore.Docs/tree/Rick-Anderson-Json-Txt-patch-3
Then clone that

Comment (Paul-Dempsey) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Hmm, my compile fix disappeared - trying again.

```
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.Formatters;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Options;

namespace WebApp1
{
    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            var jsonPatchServices = new ServiceCollection();

            jsonPatchServices.AddLogging()
                .AddMvc()
                .AddNewtonsoftJson();

            var jsonPatchOptions = jsonPatchServices.BuildServiceProvider()
                .GetRequiredService<IOptions<MvcOptions>>().Value;

            var jsonPatchInputFormatter = jsonPatchOptions.OfType<NewtonsoftJsonPatchInputFormatter>().First();

            services.AddControllersWithViews(options => jsonPatchOptions.InputFormatters.Insert(0, jsonPatchInputFormatter));
        }
```
There were missing usings, and AddNewtonsoft.Json doesn't return a type that contains BuildServiceProvider -- the fluent chain of calls mutated the type being propagated.

This still generates a warning for use of `BuildServiceProvider` (see https://stackoverflow.com/questions/31863981/how-to-resolve-instance-inside-configureservices-in-asp-net-core), but I'm out of time and the knowledge to fix that.

Comment (Paul-Dempsey) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: I wasn't able to push and make a pr because of auth issues.

Comment (Rick-Anderson) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: You'd have to PR it. That's why I find it easier to use the browser to get started.

Thanks for the help, @serpent5  sent me the code.

Is everyone happy with the changes? How about the H2 I added? Need some SEO juice the H2 will provide.

Comment (Paul-Dempsey) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: I'd be happier with code that resolves the warning:

c:\repos\AspNetCore.Docs\aspnetcore\web-api\jsonpatch\samples\3.0\WebApp1\Startup.cs(33,36): warning ASP0000: Calling 'BuildServiceProvider' from application code results in an additional copy of singleton services being created. Consider alternatives such as dependency injecting services as parameters to 'Configure'. [C:\repos\AspNetCore.Docs\aspnetcore\web-api\jsonpatch\samples\3.0\WebApp1\WebApp1.csproj]

Comment (Rick-Anderson) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: I don't get any warnings in the code I used.

Comment (Rick-Anderson) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: That is, the code @serpent5  sent me that's merged.

Comment (Paul-Dempsey) [reply aspnetcore/web-api/jsonpatch/samples/3.0/WebApp1/Startup.cs]: Merged and confirmed - I'm happy now 👍

**Pull request merged**